### ISSUE TITLE
Tab trap in file editor prevents keyboard navigation

### ISSUE THREAD

**Issue by IsabelParedes** [jupyterlab/jupyterlab#18815]
## Description

- When opening and editing a file with the file editor, the user cannot navigate outside of the editor with a keyboard.

## Reproduce

1. Create a new text file.
2. Open the file, the editor should be in focus.
3. Try to navigate outside the editor with the keyboard only, pressing `Esc` or `Tab`.
4. The editor cannot be escaped, this is a tab trap.

<img width="1568" height="576" alt="tab trap in file editor" src="https://github.com/user-attachments/assets/df67b90c-a6ee-4971-838a-a030cec6c948" />

## Expected behavior

- Users should be able to move focus away from the editor with `Esc`.

## Context

- Operating System and version: Garuda Linux
- Browser and version: Chromium v147.0.7727.116 
- JupyterLab version: v4.5.6
